### DISCUSSION TITLE
[Razer Phone 2] Android 11 install guide!

### DISCUSSION THREAD

**Discussion by Marcos8760** [phhusson/treble_experimentations#1823] [Show and tell]
Here is a thing that i made a while back on XDA (about a couple of months ago.) Just trying to get a bit more coverage so everyone can get the full potential of their RP2's.

https://forum.xda-developers.com/t/guide-android-11-gsi-with-gapps-and-root.4216117/

I also made a YT video in the guide. 

Happy flashing!

- L

**Comment by haruki-taka8** [comment]
I'm glad to see more people involved in the GSI flashing community of the Phone 2 :)


> Things that don't work for some reason: Magisk (root is available through superuser (the reason you would want magisk is for modules))

It is possible to install Magisk, see #1667. I did it on my Phone 2 a few days ago. There's also a Phone 2 flashing guide in [phhusson/treble_experimentations](https://github.com/phhusson/treble_experimentations/wiki/Razer-Phone-2) (aka the repo we're in right now).

**Reply by Marcos8760** [reply]
Every time i try to run `phh-securize.sh`, nothing happens.

**Reply by Marcos8760** [reply]
> Every time i try to run `phh-securize.sh`, nothing happens.

Looking closer, nothing is happening because even though I am running as root, I can't mount system as RW??!

**Reply by Marcos8760** [reply]
this is the face of a man who feels stupid. :| (i didn't know what the vndklite variant does...) I've spent months trying to fix this and now i feel stupid. I hope people learn from my mistakes. VNDKLITE MAKES THINGS READ/WRITE!!

**Reply by haruki-taka8** [reply]
It also took me forever to realize vndklite enables system mounting! I believe you should add this to your guide in XDA.

**Reply by Marcos8760** [reply]
I got magisk but it's not passing safteynet...  (i added literally everything to my guide. inclu. magisk guide.)

**Reply by haruki-taka8** [reply]
Magisk's built-in SafetyNet test is broken in 22.1 - so use an external checker from Play Store.

Have you tried MagiskHide and MagiskHide Props Config (#116) to pass SafetyNet yet?

**Reply by Marcos8760** [reply]
Yes (using MagiskHide and MagiskHide Props Config). Basic integrity is passed but CTS profile match isn't.

**Reply by haruki-taka8** [reply]
That's actually pretty nice already! Couldn't even get my basic integrity passed. Sorry I can't be of help regarding Magisk - it's a bit out of my scope of knowledge.

**Reply by Marcos8760** [reply]
yea same lol.

**Comment by Marcos8760** [comment]
I managed to get Magisk working on [HavocOS](https://forum.xda-developers.com/t/11-official-havoc-os-4-1-arm64-arm-a64-a-ab-ab-vndklite-gapps-vanilla.4076903/) literally just by renaming resetprop (in system/bin) to resetprop-phh, then installing.
